### ISSUE TITLE
Visual Studio: Assertion Failed: API version mismatch

### ISSUE THREAD

**Issue by crbranch** [Microsoft/TypeScript#8975]
I have a ASP.NET Core project I created with Visual Studio 2015 that includes some Typescript code.  A few days ago I updated to ASP.NET Core RC2 (uninstalled RC1, then installed RC2).  Ever since then, when I attempt to open any typescript (.ts) file in Visual Studio, Visual Studio 2015 displays an assertion error and then locks up and requires a force quit.  It happens for any .ts file, even when no project/solution is loaded.  The error message is pasted in below.  Any ideas on how to troubleshoot would be appreciated.  The Help | About screen in Visual Studio 2015 shows the Typescript version as 1.8.30.0.

![assert](https://cloud.githubusercontent.com/assets/17033045/15808120/372f812c-2b3d-11e6-8992-d3b929baf6a7.png)

**Comment by mhegazy** [comment]
have you enabled your [dev mode](https://github.com/Microsoft/TypeScript/wiki/Dev-Mode-in-Visual-Studio) before? do you have a registry entry at `HKEY_CURRENT_USER\SOFTWARE\Microsoft\VisualStudio\14.0\TypeScriptLanguageService\CustomTypeScriptServicesFileLocation`?

**Comment by crbranch** [comment]
I have not enabled **dev mode** before (at least that I know of).  The registry key you asked about does exist, the current value is:

C:\Users\Branch\AppData\Roaming\npm\node_modules\typescript\lib\typescriptServices.js

**Comment by mhegazy** [comment]
The registry entry is added by the dev mode script. so i am guessing you ran this at one point in time. looking at the version in `C:\Users\Branch\AppData\Roaming\npm\node_modules\typescript\lib\typescriptServices.js` would give you a rought estimate to when that happened.

You can either remove this registry key, and thus VS will use the JS file that is part of the plugin (with the matching API version), or you can update your custom TS location to point to a later TS package (1.8.10 or typescript@next). 

Judging by your comment, i am guessing you would want to remove the entry all together.

Here is some more details about the issue. The VS plugin is written into two parts, a managed VS plugin, and a .js file  ("typescriptServices.js"). most of the language logic is part of the JS file. The contract between the two has a version, and both should match to ensure correct behavior. installing the VS pluing installs the VS part, and the matching JS file. 
For users wanting to try latest version of TS (e.g. nighly version of the package in typescript@next), they can point VS to load the JS file from a custom location. this is done thorough setting the registry key above. this takes precedence over the one installed with the package.

**Comment by crbranch** [comment]
@mhegazy Removing the registry entries does resolve the problem.  Thank you for your help, and for the additional explanation regarding how the pieces fit together.

Apparently I did run the VSDevMode.ps1 script at some point back in early February, though I don't recall the exact details.  I do recall I was having issues described in #6958 after updating the Typescript compiler back then, so I may have tried VSDevMode.ps1 while looking for the solution to that issue.  In any case, I deleted the following registry values that were apparently added by VSDevMode.ps1, and everything seems to be working now:

CustomDefaultES6LibraryLocation
CustomDefaultLibraryLocation
CustomTypeScriptServicesFileLocation
